### PR TITLE
machine/microbit-v2: add initial support for microbit v2 board

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds initial support for the Microbit v2 board: https://microbit.org/new-microbit/

Much of the work is based on work started by @alankrantas thank you.

### CODE REVIEW

**Comment by deadprogram** [comment]
Now also tested I2C interface with built-in `lsm303agr` and is working.

File: tools/gen-device-svd/gen-device-svd.go [hunk 1]
```diff
@@ -537,7 +537,14 @@ func parseBitfields(groupName, regName string, fieldEls []*SVDField, bitfieldPre
 				enumName = strings.ToUpper(enumName)
 			}
 			enumDescription := strings.Replace(enumEl.Description, "\n", " ", -1)
-			enumValue, err := strconv.ParseUint(enumEl.Value, 0, 32)
```
Comment (aykevl) [inline tools/gen-device-svd/gen-device-svd.go:540]: I see, this doesn't work in Go before version 1.13 due to binary literals being new?

**Review by aykevl — APPROVED** [review]
Looks good to me, thank you!

**Comment by aykevl** [comment]
@deadprogram I think you intended this PR to be squased to one commit? Is that correct? (If so, feel free to do so)

**Comment by deadprogram** [comment]
I rebased into 3 commits:
- one with the updated submodule
- one with the changes to the generator
- one with the new Microbit v2 board itself

I will merge once it passes the CI.

File: tools/gen-device-svd/gen-device-svd.go (hunk 1 above)
Comment (deadprogram) [reply tools/gen-device-svd/gen-device-svd.go:540]: Correct.

**Pull request merged**